### PR TITLE
WIP: Added Outlines logits processor for JSON schema validation

### LINKED ISSUE #176: Add logits processor
### Feature request

Add the possibility for users to specify a function that processes the logits before sampling. This function would be called right before `next_token_chooser`.

### Motivation

I am a maintainer of [Outlines](https://github.com/outlines-dev/outlines) a library for guided generation: regex, JSON, grammar, etc. A user [recently asked](https://github.com/outlines-dev/outlines/discussions/523) if we could integrate with LoRaX, and I think it would be beneficial for both libraries.

We can then discuss a potential deeper integration (allow function calling via JSON-guided generation), but this feature request is minimally intrusive and low maintenance.

### Your contribution

I could help with submitting a PR.

### PR DESCRIPTION
**Pull request by tgaddair**
Closes #176

### CODE REVIEW

**Review by tgaddair — COMMENTED** [review]
LGTM!
